repose-aggregator/core/repose-core/src/main/java/org/openrepose/core/services/rms/ResponseMessageServiceImpl.java

```diff
@@ -93,10 +96,19 @@ public boolean isInitialized() {
     }
 
     @Override
-    public void handle(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    public void handle(HttpServletRequest request, HttpServletResponseWrapper response) throws IOException {
         final StatusCodeMatcher matchedCode = getMatchingStatusCode(String.valueOf(response.getStatus()));
-        final MutableHttpServletRequest mutableRequest = MutableHttpServletRequest.wrap(request);
-        MediaRangeProcessor processor = new MediaRangeProcessor(mutableRequest.getPreferredHeaders("Accept", DEFAULT_TYPE));
+        final HttpServletRequestWrapper wrappedRequest = new HttpServletRequestWrapper(request);
```

Comment:
Since you are already passing in the wrapped response, why not go ahead and pass in the wrapped request, instead of immediately turning around and wrapping it?